### PR TITLE
Expose additional netstandard APIs in Sockets.

### LINKED ISSUE #14182: Missing some Socket API
These are missing
```
M:System.Net.Sockets.Socket.EndAccept(System.Byte[]@,System.IAsyncResult)
M:System.Net.Sockets.Socket.EndAccept(System.Byte[]@,System.Int32@,System.IAsyncResult)
P:System.Net.Sockets.SocketAsyncEventArgs.SendPacketsFlags
M:System.Net.Sockets.SocketAsyncEventArgs.get_SendPacketsFlags
M:System.Net.Sockets.SocketAsyncEventArgs.set_SendPacketsFlags(System.Net.Sockets.TransmitFileOptions)
```

Broken out of https://github.com/dotnet/corefx/issues/12460

@karelz This needs an owner.

### PR DESCRIPTION
**Pull request by Priya91**
cc @karelz @danmosemsft @CIPop @stephentoub 

Added tests for SendPacketsFlags. EndAccept methods are already tested through previously exposed EndAccept.

fixes #14182

### CODE REVIEW

File: src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs [hunk 1]
```diff
@@ -279,6 +279,26 @@ public void BufferZeroCountThenNormal_ZeroCountIgnored(SocketImplementationType
...
+        }
+
+        [Theory]
+        [InlineData(SocketImplementationType.APM)]
+        [InlineData(SocketImplementationType.Async)]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void SocketDisconnectedAndReusable_TransmitFileOptionReuseSocket(SocketImplementationType type)
+        {
+            SendPackets(type, new SendPacketsElement(new byte[10], 4, 4), TransmitFileOptions.Disconnect | TransmitFileOptions.ReuseSocket, 4);
+        }
```
Comment (stephentoub) [inline src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs]: Do we have/need tests that validate behavior on Unix?

Comment (Priya91) [inline src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs]: We dont need, SendPacketsAsync throws PNSE on Unix..

Comment (stephentoub) [inline src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs]: > We don't need

Shouldn't we have test that verifies that?

Comment (Priya91) [inline src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs]: The test for that already exists.. Source code [here](https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs#L61)

**Pull request merged**